tasks/build/os_shell_scripts.js

```diff
@@ -0,0 +1,41 @@
+import {unlink} from 'fs';
+import {join, extname} from 'path';
+import {promisify} from 'bluebird';
+import {ncp} from 'ncp';
+const pncp = promisify(ncp);
+const punlink = promisify(unlink);
+
+module.exports = function (grunt) {
+  grunt.registerTask('_build:osShellScripts', function () {
+    const done = this.async();
+    const source = 'build/kibana/bin';
+    const platforms = grunt.config.get('platforms');
+
+    Promise.resolve()
+      .then(() => allPlatforms(platform => punlink(join(platform.buildDir, 'bin')))
+      .then(() => allPlatforms(platform => pncp(source, join(platform.buildDir, 'bin'))))
+      .then(() => allPlatforms(removeExtraneousShellScripts)))
+      .then(() => done)
+      .catch(err => done(err));
+
+    function allPlatforms(fn) {
```

Comment:
This is entirely a personal preference, so feel free to disregard it if you disagree, but I think it's best if actual function logic exists in standalone functions that do not close over values from another function.  You could use partial application inside the task function to close over `platforms` if you still want the calls to `allPlatforms` to be concise.

In other words, something like this:

```js
grunt.registerTask(/* ... */, () => {
  // ... stuff
  const platforms = grunt.config.get('platforms');
  const allPlatforms = (fn) => invokeAllAsync(platforms, fn);
});

function invokeAllAsync(arr, fn) {
  return Promise.all(arr.map(fn));
}
```

This may be a little overkill in this particular situation, but it helps separate testable logic from the actions we take to improve code legibility.